nameable.rb

```diff
@@ -0,0 +1,30 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Not Implmented'
+  end
+end
+
+class Decorator < Nameable
+  attr_accessor :name
```

Comment:
- [ ] @tarikbouari great work in your class however your accessor attribute should be nameable since you don't use name in your class.